rosbag2_cpp/test/rosbag2_cpp/test_sequential_writer.cpp

```diff
@@ -1162,6 +1302,56 @@ TEST_P(ParametrizedTemporaryDirectoryFixture, split_bag_metadata_has_full_durati
   ASSERT_EQ(metadata.duration, std::chrono::nanoseconds(500));
 }
 
+TEST_F(SequentialWriterTest, filename_extracts_prefix_from_timestamped_directory_name)
+{
+  // Test that timestamp pattern is removed from directory name when generating filename
+  // This simulates the case where:
+  // * --output is not specified
+  // * default timestamped directory is used
+  const std::string timestamped_dir = "rosbag2_2025_11_04-20_21_42";
+  storage_options_.uri = (tmp_dir_ / timestamped_dir).string();
+
+  // Set up metadata interception
+  ON_CALL(*metadata_io_, write_metadata).WillByDefault(
+    [this](const std::string &, const rosbag2_storage::BagMetadata & metadata) {
+      fake_metadata_ = metadata;
+    });
+  EXPECT_CALL(*metadata_io_, write_metadata).Times(1);
+
+  auto sequential_writer = std::make_unique<rosbag2_cpp::writers::SequentialWriter>(
+    std::move(storage_factory_), converter_factory_, std::move(metadata_io_));
+  writer_ = std::make_unique<rosbag2_cpp::Writer>(std::move(sequential_writer));
+
+  std::string rmw_format = "rmw_format";
+  writer_->open(storage_options_, {rmw_format, rmw_format});
```

Comment:
Can be simplified to be as:
```suggestion
  writer_->open((tmp_dir_ / timestamped_dir).string());
```